Gradle OOM error across different platforms

For example: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8775032916537102849/+/u/run_test.dart_for_build_tests_shard_and_subshard_3_3/test_stdout

```
[        ] Execution failed for task ':app:packageDebug'.
[        ] > A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable
[        ]    > java.lang.OutOfMemoryError (no error message)
```

and

```
[        ] Caused by: java.lang.OutOfMemoryError: Java heap space
[        ] 	at java.base/java.io.ByteArrayOutputStream.<init>(ByteArrayOutputStream.java:81)
[        ] 	at com.android.zipflinger.NoCopyByteArrayOutputStream.<init>(NoCopyByteArrayOutputStream.java:27)
[        ] 	at com.android.zipflinger.Compressor.deflate(Compressor.java:35)
[        ] 	at com.android.zipflinger.BytesSource.build(BytesSource.java:89)
[        ] 	at com.android.zipflinger.BytesSource.<init>(BytesSource.java:50)
[        ] 	at com.android.zipflinger.Sources.from(Sources.java:40)
[        ] 	at com.android.zipflinger.Sources.from(Sources.java:32)
[        ] 	at com.android.builder.internal.packaging.ApkFlinger$writeFile$1.call(ApkFlinger.kt:198)
[        ] 	at com.android.builder.internal.packaging.ApkFlinger$writeFile$1.call(ApkFlinger.kt:195)
[        ] 	at java.base/java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1428)
[        ] 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
[        ] 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
[        ] 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
[        ] 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
[        ] 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

I have been seeing this intermittently all week.

--- a/dev/integration_tests/flutter_gallery/android/gradle.properties
+++ b/dev/integration_tests/flutter_gallery/android/gradle.properties
@@ -1,3 +1,3 @@
-org.gradle.jvmargs=-Xmx1536M
+org.gradle.jvmargs=-Xmx4G
 android.useAndroidX=true
 android.enableJetifier=true

